moveit_core/robot_model/src/robot_model.cpp

```diff
@@ -1290,6 +1290,8 @@ double RobotModel::distance(const double* state1, const double* state2) const
 
 void RobotModel::interpolate(const double* from, const double* to, double t, double* state) const
 {
+  ROS_WARN_STREAM_COND_NAMED(std::isnan(t) || t < 0. || t > 1., LOGNAME,
```

Comment:
Would it make sense to add `std::isinf` check too? Also, typo in word `parameter`.

Reply:
Actually, should this whole function just fail if the fraction `t` is not reasonable? Print error and return instead of giving a warning? After all, there was a crash when `t` was `NaN`. :thinking:

Reply:
I believe infinite values will be caught by `t < 0. || t > 1.`. Actually, I think a NaN would also, but I'm not sure that's true across all platforms, etc. Thanks for the spelling catch.

Reply:
I was on the fence about how strongly it should react to an incorrect value. Since it's a `void` return type, failure isn't really an option, just complaining and doing nothing. (Or, maybe, throwing an exception?)

Reply:
They should probably throw an exception. Interpolating from NaNs and infs sounds dangerous. What about an `assert`?

Reply:
Values smaller than 0 and larger than 1 are perfectly fine: You just extrapolate beyond the point. However, a warning in this case is probably useful. `nan` and `inf` could throw.

Reply:
> Values smaller than 0 and larger than 1 are perfectly fine: You just extrapolate beyond the point. However, a warning in this case is probably useful. `nan` and `inf` could throw.

Thumbs up for this approach.